vmm/src/device_manager.rs

```diff
@@ -3708,6 +3711,12 @@ impl DeviceManager {
         &mut self,
         device_cfg: &mut DeviceConfig,
     ) -> DeviceManagerResult<PciDeviceInfo> {
+        if let Some(id) = &device_cfg.id {
+            if self.identifier_not_unique(id) {
```

Comment:
Is this necessary? We call `VmConfig::validate()` on the patched config before even getting into this code.

Reply:
You're right, because of the previous commit and because `VmConfig::validate()` is already called from `add_...()`, we're already covered. I'll remove this commit.